lib/injection_container.dart

```diff
@@ -0,0 +1,90 @@
+import 'layers/domain/use_cases/subscribe_users_use_case.dart';
+import 'layers/domain/use_cases/set_realm_pref_use_case.dart';
+import 'layers/domain/use_cases/subscribe_realm_use_case.dart';
+import 'layers/domain/use_cases/subscribe_theme_use_case.dart';
+import 'layers/local/data_sources/local_source_impl/local_datasource_impl.dart';
+import 'layers/local/data_sources/sources/drift_database/database/database.dart';
+import 'layers/presentation/settings_page/bloc/settings_cubit.dart';
+import 'layers/presentation/statistic_page/widgets/achieves_widget/bloc/achieves_data_cubit.dart';
+import 'layers/presentation/statistic_page/widgets/personal_data_widget/bloc/personal_data_cubit.dart';
+import 'layers/presentation/statistic_page/widgets/vehicles_widget/bloc/vehicles_data_cubit.dart';
+import 'layers/remote/remote_source_impl/remote_source_impl.dart';
+import 'layers/remote/sources/wot_api_client.dart';
+
+final inj = GetIt.instance;
+
+Future<void> init() async {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  inj.registerFactory(
+      () => SettingsCubit(subscribeTheme: inj(), setTheme: inj()));
+
+  inj.registerFactory(() => SingInCubit(
+        saveUser: inj(),
+        subscribeUsers: inj(),
+        subscribeRealm: inj(),
+        setRealm: inj(),
+        removeUserUseCase: inj(),
+        signIn: inj(),
+      ));
+
+  inj.registerFactory(() => PersonalDataCubit(loadData: inj()));
+  inj.registerFactory(() => VehiclesDataCubit(loadVehicles: inj()));
+  inj.registerFactory(() => AchievesDataCubit(loadAchieves: inj()));
+
+  inj.registerFactory(() => LoadAchievesData(repository: inj()));
+  inj.registerFactory(() => LoadVehiclesData(repository: inj()));
+  inj.registerFactory(() => SubscribeThemeUseCase(repository: inj()));
+  inj.registerFactory(() => SaveUserUseCase(repository: inj()));
+  inj.registerFactory(() => SubscribeUsers(repository: inj()));
+  inj.registerFactory(() => SubscribeRealm(repository: inj()));
+  inj.registerFactory(() => SetRealmUseCase(repository: inj()));
+  inj.registerFactory(() => RemoveUserUseCase(repository: inj()));
+  inj.registerFactory(() => SetThemeUseCase(repository: inj()));
+  inj.registerFactory(() => LoadPersonalData(repository: inj()));
+  inj.registerFactory(() => SingInUseCase(repository: inj()));
+
+  inj.registerLazySingleton<Repository>(() => RepositoryImpl(
+      localSource: inj(), remoteSource: inj(), baseOptions: inj()));
+
+  inj.registerFactory<RemoteDataSource>(
+      () => RemoteSourceImpl(wotClient: inj()));
+
+  inj.registerFactory<LocalDataSource>(() => LocalDataSourceImpl(
+        sharedPreferences: inj(),
+        wotStatDao: inj(),
+      ));
+
+  inj.registerFactory<WotStatDao>(() => WotStatDao(inj()));
+
+  inj.registerFactory<WotClient>(() => WotClient(inj()));
+  final SharedPreferences sharedPref = await SharedPreferences.getInstance();
+  final WotStatDatabase driftDatabase = constructDb();
+  final BaseOptions baseOptions = BaseOptions();
```

Comment:
It is a bit strange bacause you creating `BaseOptions` within `init` method which supposed to be called ones, but asking GetIt to create new instance each time, but it will return the same instance each time. Moreover looks like BaseOptions should be singleton.